### PR TITLE
Puma: Bulk input component improvements

### PR DESCRIPTION
**Pull request by lor11212**
Issue: https://devtopia.esri.com/Zurich-R-D-Center/ce-plugin-crew/issues/42

This PR implements the improvements suggested in #70:
- Added a warning message in case an input doesn't match the key:value pattern. With branch and line numbers.
- Added a warning message in case a line is empty to inform it has been ignored.
- Added parenthesis to the component displayed name: "CityEngine Puma (bulk)".

image

These changes should help debugging issues caused by wrong cga attributes more easily.

### CODE REVIEW

File: PumaGrasshopper/ComponentPumaBulkInput.cs [hunk 1]
```diff
@@ -130,27 +130,42 @@ private RuleAttributesMap ParseBulkInputTree(IGH_DataAccess DA, int shapeCount)
...
+                                    throw new Exception("Attribute key " + pair.Key + " has no corresponding value");
+                                }
                             }
                         }
-                    });
+                        else if(pair.Key.Length == 0)
+                        {
+                            AddRuntimeMessage(GH_RuntimeMessageLevel.Warning,
+                                              "Branch " + branchId + " - Line " + i + ": Ignoring empty line.");
+                        }
```
Comment (mistafunk) [inline PumaGrasshopper/ComponentPumaBulkInput.cs:162]: I suggest to remove this warning, i.e. silently ignore empty lines.

**Review by mistafunk — CHANGES_REQUESTED** [review]
Overall looks good.

Please avoid the warnings for empty lines, it can be useful for the user to organize a text file with empty lines (group the attributes). Please also add a line to the doc, that empty lines are silently ignored.

File: PumaGrasshopper/ComponentPumaBulkInput.cs (hunk 1 above)
Comment (lor11212) [reply PumaGrasshopper/ComponentPumaBulkInput.cs]: I did the changes you requested. Ready for re-review 👍

**Review by mistafunk — APPROVED** [review]

File: PumaGrasshopper/ComponentPumaBulkInput.cs (hunk 1 above)
Comment (mistafunk) [reply PumaGrasshopper/ComponentPumaBulkInput.cs]: Looks good

**Pull request merged**